x/data/types.go

```diff
@@ -26,27 +26,43 @@ func (ch ContentHash) Validate() error {
 	return sdkerrors.ErrInvalidRequest.Wrapf("content hash must be one of raw type or graph type")
 }
 
-func (chr ContentHash_Raw) Validate() error {
-	err := chr.DigestAlgorithm.Validate(chr.Hash)
+func (chr *ContentHash_Raw) Validate() error {
+	err := validateHash(chr.Hash, chr.DigestAlgorithm)
 	if err != nil {
 		return err
 	}
 
-	return chr.MediaType.Validate()
+	ext := chr.FileExtension
+	extLen := len(ext)
+	if extLen < 2 {
+		return sdkerrors.ErrInvalidRequest.Wrapf("file extension cannot be shorter than 2 characters")
+	}
+
+	if extLen > 6 {
+		return sdkerrors.ErrInvalidRequest.Wrapf("file extension cannot be longer than 6 characters")
+	}
+
+	// check that ext is all lowercase or numeric
+	for _, c := range ext {
+		if c < '0' || c > '9' && c < 'a' || c > 'z' {
+			return sdkerrors.ErrInvalidRequest.Wrapf("file extension must be all lowercase or numeric")
```

Comment:
can we add a test case for this in iri_test.go?

Reply:
Added a case